### ISSUE TITLE
Issues dumping in VS2022 17.8

### ISSUE THREAD

**Issue by Dessyreqt** [ycherkes/ObjectDumper#68]
![image](https://github.com/ycherkes/ObjectDumper/assets/12574786/e77b0b01-c8b3-4f10-8d61-1d45c5f24fd4)
The data I'm trying to dump is coming from the database in fixed-width `char` columns. To reproduce this error, here is some code:

```C#
var testObj = new TestObj
{
    Name = "Test".PadRight(50),
    Id = Guid.NewGuid(),
    GroupId = "Group Id".PadRight(50),
    ParentGroup = "Parent Group".PadRight(12),
    GroupName = "Group Name".PadRight(100),
    ObjectType = "Object Type",
    IsObject = true,
    IsActive = true,
    CreateDate = DateTime.Now,
    CreateUser = "Create User".PadRight(50),
    ObjectCode = null,
    IsOwned = true,
    IsValid = true,
    IsStandard = true,
    Description = "Description"
};

testObj.ToString(); // set breakpoint here to dump testObj while debugging

class TestObj
{
    public string? Name { get; set; }
    public Guid Id { get; set; }
    public string GroupId { get; set; } = string.Empty;
    public string? ParentGroup { get; set; }
    public string? GroupName { get; set; }
    public string ObjectType { get; set; } = string.Empty;
    public bool IsObject { get; set; }
    public bool IsActive { get; set; }
    public DateTime CreateDate { get; set; }
    public string CreateUser { get; set; }
    public string? ObjectCode { get; set; }
    public bool? IsOwned { get; set; }
    public bool? IsValid { get; set; }
    public bool? IsStandard { get; set; }
    public string? Description { get; set; }
}
```

I am on version 17.8 of VS2022. At first I thought this issue was due to the padded fields (since the error makes mention of greater than two padded characters) but I can just comment out either the bottom or top half of the object initializer and the dump works fine, even including the extra spaces in the strings that I left uncommented. Additionally, even if I go and Trim every string that has padding, I run into this issue when trying to dump a list of 10 such objects.

I have not run into any similar issues in 17.7.x version of VS, even with larger objects or similar objects or lists of objects. So I suspect there was a change in VS2022 17.8 that is affecting the operation of this extension.

**Comment by ycherkes** [comment]
Hi, thanks for reporting.

[Debugger.GetExpression](https://learn.microsoft.com/en-us/dotnet/api/envdte.debugger.getexpression?view=visualstudiosdk-2022#examples) returns truncated result starting from version 17.8.
Seems, the max length of the data is 1024 symbols:

![image](https://github.com/ycherkes/ObjectDumper/assets/13467759/81d96cfa-0dd1-4108-99e2-171efaccd6ec)

I will try to bypass this behavior.

**Comment by ycherkes** [comment]
Opened a [Visual Studio issue](https://developercommunity.visualstudio.com/t/VS-1780-Debugger-trims-expression-eval/10518078)

**Comment by ycherkes** [comment]
Hi @Dessyreqt,
I have just published version 89 as a temporary solution. See [PR#69](https://github.com/ycherkes/ObjectDumper/pull/69).
Please check if it works for you.
I would prefer to leave this issue open pending Microsoft's response.

**Comment by Dessyreqt** [comment]
The update is working for my use case. Thanks.

**Comment by ycherkes** [comment]
Fixed in v 0.0.0.90
